### PR TITLE
Add missing ARNs for AWS resources

### LINKED ISSUE #935: Add missing ARNs
https://github.com/nccgroup/ScoutSuite/issues/511 has been mostly implemented but the following still lack ARNs:

- [ ] cloudwatch/metric_filters.py
- [ ] directconnect/connections.py
- [ ] ec2/vpcs.py
- [ ] efs/filesystems.py
- [ ] elasticache/parametergroups.py
- [ ] elasticache/securitygroups.py
- [ ] elasticache/subnetgroups.py
- [ ] elasticache/vpcs.py
- [ ] elb/policies.py
- [ ] elb/vpcs.py
- [ ] elbv2/vpcs.py
- [ ] emr/clusters.py
- [ ] emr/vpcs.py
- [ ] iam/passwordpolicy.py
- [ ] kms/grants.py
- [ ] rds/subnetgroups.py
- [ ] rds/vpcs.py
- [ ] redshift/cluster_parameters.py
- [ ] redshift/cluster_security_groups.py
- [ ] redshift/vpcs.py
- [ ] route53/hosted_zones.py
- [ ] ses/identity_policies.py
- [ ] sns/subscriptions.py
- [ ] vpc/flow_logs.py
- [ ] vpc/network_acls.py
- [ ] vpc/peering_connections.py
- [ ] vpc/subnets.py
- [ ] vpc/vpcs.py

Additionally where we build ARNs ourselves (since the API doesn't return a value), we should use a normalized static function, the string building is error prone.

### PR DESCRIPTION
**Pull request by lowSoA**
# Description

The following resources lacked ARN processing and being displayed in their corresponding partials:

* __cloudwatch/metric__filters.py__: Added ARN processing and display in corresponding partial
* __directconnect/connections.py__: Added ARN processing but couldn't find the corresponding partial to display it
* __ec2/vpcs.py__: No need to update since it uses a superclass which has the updated ARN processing
* __efs/filesystems.py__: Added ARN processing but couldn't find the corresponding partial to display it
* __elasticache/parametergroups.py__: Already has the correct format and is displayed
* __elasticache/subnetgroups.py__: Already has the correct format and is displayed
* __elasticache/vpcs.py__: No need to update since it uses a superclass which has the updated ARN processing
* __elb/policies.py__: Added ARN processing and display in corresponding partial
* __elb/vpcs.py__: No need to update since it uses a superclass which has the updated ARN processing
* __elbv2/vpcs.py__: No need to update since it uses a superclass which has the updated ARN processing
* __emr/clusters.py__: Added ARN processing and display in corresponding partial
* __emr/vpcs.py__: Added ARN processing and display in corresponding partial
* __kms/grants.py__: Added ARN processing and display in KMS keys partial
* __rds/subnetgroups.py__: Added ARN processing and display in corresponding partial
* __rds/vpcs.py__: No need to update since it uses a superclass which has the updated ARN processing
* __redshift/cluster__parameters.py__: Added ARN processing and display in corresponding partial
* __redshift/vpcs.py__: No need to update since it uses a superclass which has the updated ARN processing
* __route53/hosted__zones.py__: Added ARN processing and display in corresponding partial
* __ses/identity__policies.py__: Added ARN processing and display in corresponding partial (also fixed display issue in partial)
* __sns/subscriptions.py__: Added ARN processing and display in SNS topics partial (also fixed display issue in partial)
* __vpc/flow__logs.py__: Added ARN processing and display in corresponding partial
* __vpc/network__acls.py__: Added ARN processing and display in corresponding partial
* __vpc/peering__connections.py__: Added ARN processing and display in corresponding partial
* __vpc/subnets.py__: Added ARN processing and display in corresponding partial
* __vpc/vpcs.py__: No need to update since it uses a superclass which has the updated ARN processing

Other missing ARNs which were not updated:
* __iam/passwordpolicy.py__: Could not find any _identifiable_ piece of information to "generate" an ARN
* __elasticache/securitygroups.py__: Recently created AWS accounts don't allow network level resources to be created outside of a VPC ([reference](https://docs.aws.amazon.com/AmazonElastiCache/latest/mem-ug/SecurityGroups.html))
* __redshift/cluster__security__groups.py__: Same as above

Fixes #935 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (optional)
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: ScoutSuite/providers/aws/utils.py [hunk 1]
```diff
@@ -111,3 +111,27 @@ def snake_keys(d):
...
+    :param account_id:                  The ID of the AWS account that owns the resource
+    :param resource_id:                 The resource identified
+    :param resource_type:               (Optional) The resource type
+    :return:                            Resource ARN
+    """
+
+    try:
+        # If a resource type is specified
+        if resource_type is not None:
+            arn = 'arn:{}:{}:{}:{}:{}/{}'.format(partition, service, region, account_id, resource_type, resource_id)
```
Comment (j4v) [inline ScoutSuite/providers/aws/utils.py:131]: We should try to use f-Strings wherever possible ;-)

https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python

File: ScoutSuite/providers/aws/utils.py [hunk 2]
```diff
@@ -111,3 +111,27 @@ def snake_keys(d):
             else:
                 new_table[new_key] = d[k]
     return new_table
+
+def format_arn(partition, service, region, account_id, resource_id, resource_type=None):
```
Comment (j4v) [inline ScoutSuite/providers/aws/utils.py:115]: Noice

File: ScoutSuite/providers/aws/utils.py (hunk 1 above)
Comment (lowSoA) [reply ScoutSuite/providers/aws/utils.py]: Ah, my bad, old habits die hard :) Fixed in [the latest commit](https://github.com/nccgroup/ScoutSuite/pull/1022/commits/ab3100b51927fd3d1e0366b9d8137b0fa45e5c18)

**Pull request merged**